content/chainguard/migration/migration-tips.md

```diff
@@ -0,0 +1,244 @@
+
+```bash
+docker run -it python echo "in a shell"
+in a shell
+```
+
+The Chainguard Python image doesn't use an entrypoint script. It relies on the Python interpreter as single entrypoint for both the `latest` and the `latest-dev` variants. So instead of executing the command through a shell, it tries to parse the command as an argument to the Python interpreter:
+
+```bash
+docker run -it cgr.dev/chainguard/python echo "in a shell"
+/usr/bin/python: can't open file '//echo': [Errno 2] No such file or directory
+```
+
+The same behavior can be observed in the `latest-dev` variant, which does contain a shell, but uses the Python interpreter as entrypoint to keep consistency with the `latest` variant:
+
+```bash
+docker run -it cgr.dev/chainguard/python:latest-dev echo "in a shell"
+/usr/bin/python: can't open file '//echo': [Errno 2] No such file or directory
+```
+
+Other images, such as our [WordPress container images](https://images.chainguard.dev/directory/image/wordpress/overview?utm_source=cg-academy&utm_medium=referral&utm_campaign=dev-enablement&utm_content=edu-content-chainguard-migration-migrations-overview), will have a different entrypoint behavior in their `-dev` variant to allow for customization and to facilitate migration from other base images. It's important to always read the image's documentation to understand how the entrypoint works, and if there are any major differences from other images you may be used to work with.
+
+## Containers Don't Run As root By Default
+
+Although there are exceptions, Chainguard Containers typically don’t run as the root user. The reason for this is that distroless containers should have no privileged capabilities, and containers that run as a non-root user and use a minimal seccomp profile are ideal from a security perspective.
+
+Because they don't run as the root user, you may need to include a `USER root` statement in your Dockerfile before installing software on a Chainguard Container.
+
+Additionally, be aware that `-dev` images also do not run as root in most cases, which can result in permission errors like the following:
+
+```bash
+docker run -it --entrypoint bin/bash chainguard/python:latest-dev
+
+bash-5.2$ mkdir test
+mkdir: can't create directory 'test': Permission denied
+
+bash-5.2$ sudo mkdir test
+bash: sudo: command not found
+```
+
+In cases like this, you can instead run a command like the following to access the container's shell as the root user:
+
+```bash
+docker run -it --user root --entrypoint bin/bash chainguard/python:latest-dev
+```
+
+Here, the `--user` option tells Docker to assume the root user role.
+
+
+## Packages Not Found
+
+Container images are usually meant to support every possible use case. Because of this, they often contain packages that aren't necessary for many use cases, which increases the container image's attack surface and makes it more likely to contain CVEs.
```

Comment:
I might have written this but I think it could be clearer.

Reply:
I made a slight tweak that I think makes it clearer